celery_serverless/invoker.py

```diff
@@ -1,58 +1,135 @@
 # coding: utf-8
-import os
-from pathlib import Path
+from __future__ import unicode_literals, absolute_import
 
+import functools
+import logging
+import codecs
+import json
+from pprint import pformat
+from io import BytesIO
+
+import dirtyjson
 import click
-from ruamel.yaml import YAML
-yaml = YAML()
+from celery_serverless.config import get_config
+
+
+logger = logging.getLogger(__name__)
+
+try:
+    import boto3
+    import botocore
+    try:
+        lambda_client = boto3.client('lambda')
+    except botocore.exceptions.NoRegionError:
+        logger.warning("'boto3' invoker cannot be used: please set a default region on serverless.yml")
+        lambda_client = None
+except ImportError:  # Boto3 is an optional extra on setup.py
+    lambda_client = None
 
 from .cli_utils import run
 
+
 CELERY_HANDLER_PATH = 'celery_serverless.handler_worker'
 
 
 def invoke_main(strategy=''):
-    local_serverless = './serverless.yml'
-    if not os.path.exists(local_serverless):
-        raise RuntimeError("No serverless.yml detected. Please run 'celery serverless init' to create one")
-    config = yaml.load(Path(local_serverless))
-
+    config = get_config()
     if not strategy:
         strategy = _infer_strategy(config)
 
+    logger.info("Invoke strategy selected: '%s'", strategy)
     if strategy == 'serverless':
         invoker = _invoke_serverless
     elif strategy == 'boto3':
         invoker = _invoke_boto3
     else:
         raise NotImplementedError("Could not find a way to invoke via '%s' strategy" % strategy)
 
-    return invoker(config)
+    try:
+        logs = invoker(config)  # Should raise exception on some problem
+    except RuntimeError as err:
+        if logger.getEffectiveLevel() <= logging.INFO:
+            logger.info('Invocation failed via "%s": %s', strategy, err.details)
+        raise
+    return True
+
 
 def _infer_strategy(config):
-    #if config['provider']['name'] == 'aws':
-    #    return 'boto3'
+    if config['provider']['name'] == 'aws':
+        if lambda_client:
+            return 'boto3'
+        else:
+            logger.warning("Invoke strategy 'boto3' could not be used. Falling back to 'serverless'")
     return 'serverless'
 
 
 def _invoke_serverless(config, local=False):
-    name = _get_lambda_name(config)
+    name = _get_serverless_name(config)
     command = 'serverless invoke'
     if local:
         command += ' local'
 
-    command += ' --verbose --color --function %s' % name
-    for line, retcode in run(command):
-        click.echo(line, nl=False)
+    logger.debug("Invoking via 'serverless'")
+    command += ' --log --verbose --function %s' % name
+    output = BytesIO()
```

Comment:
As we are dealing with `BytesIO()` only, we could keep a `str` object in memory. This would avoid using a pattern of 
```
if logger.getEffectiveLevel() <= logging.DEBUG:
    output.seek(0)
```
in multiple parts of the code.